application/src/main/scala/com/azavea/franklin/crawler/StacImport.scala

```diff
@@ -254,7 +254,20 @@ class StacImport(val catalogRoot: String, serverHost: NonEmptyString) {
       _       <- IO { println("Read catalog") }
       collections <- catalog.links
         .filter(_.rel == StacLinkType.Child)
-        .traverse(c => readCollectionWrapper(makeAbsPath(catalogRoot, c.href), None))
+        .traverse { c =>
+          val path = makeAbsPath(catalogRoot, c.href)
+          // lets check if it is a collection
+          readPath[StacCollection](path).attempt.flatMap {
```

Comment:
Is there a better to check if its not a collection rather than trying to decode it and fail?

Reply:
Ohh, this could be resolved by validation probably.

Reply:
I'm having trouble reasoning about what the data looks like in Franklin for nested catalogs. Do we skip the nested catalogs and assign everything to some parent collection?

Reply:
@notthatbreezy yes; I'll make this recursive so it would go until it would find the first collection.